src/DataObjects/QueuedJobDescriptor.php

```diff
@@ -432,8 +433,8 @@ public function getCMSFields()
                 )
             ),
             $jobTitle = TextField::create('JobTitle', 'Title'),
-            $status = DropdownField::create('JobStatus', 'Status', array_combine($statuses ?? [], $statuses ?? [])),
-            $jobType = DropdownField::create('JobType', 'Queue type', $this->getJobTypeValues()),
+            $status = $this->buildJobStatusField(),
+            $jobType = $this->buildJobTypeField(),
```

Comment:
Rather than duplicate the Dropdown creation logic below, I figured it made more sense to create private functions to keep everything aligned.